velox/functions/prestosql/IPAddressFunctions.h

```diff
@@ -15,14 +15,64 @@
  */
 #pragma once
 
+#include "velox/functions/Macros.h"
+#include "velox/functions/Registerer.h"
 #include "velox/functions/prestosql/types/IPAddressType.h"
 #include "velox/functions/prestosql/types/IPPrefixType.h"
 
 namespace facebook::velox::functions {
 
+template <typename T>
+struct IPPrefixFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<IPPrefix>& result,
+      const arg_type<IPAddress>& ip,
+      const arg_type<int64_t>& prefixBits) {
+    folly::ByteArray16 addrBytes;
+    memcpy(&addrBytes, &ip, ipaddress::kIPAddressBytes);
+    std::reverse(addrBytes.begin(), addrBytes.end());
+
+    result = makeIPPrefix(folly::IPAddressV6(addrBytes), prefixBits);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<IPPrefix>& result,
+      const arg_type<Varchar>& ipString,
+      const arg_type<int64_t>& prefixBits) {
+    auto tryIp = folly::IPAddress::tryFromString(ipString);
+    if (tryIp.hasError()) {
+      VELOX_USER_FAIL("Invalid IP Invalid IP address binary:{}", ipString);
+    }
+
+    result = makeIPPrefix(
+        folly::IPAddress::createIPv6(folly::IPAddress(tryIp.value())),
+        prefixBits);
+  }
```

Comment:
Presto has check for the prefixBits.
Do we check it anywhere?

```
SELECT ip_prefix(CAST('192.168.255.255' AS IPADDRESS), 199);
Query 20241210_183104_21191_chfc9 failed: IPv4 subnet size must be in range [0, 32]
```

If we introduce the check, it might be good to make a stateful `IPPrefixFunction` for const `prefixBits` and non-const for non-const one.

Reply:
I think we can add the checks as part of the stateless functions, let me add them in and update the UT

Reply:
Stateless would be executing IF() per row, but we can have it like that until it is prominent on profiling. :)